Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+
+    def _to_numpy(self, X=False, Y=False, meta=False, writable=False):
+        """Export a transformed numpy matrix.
+
+        The order is always X, Y, meta. Always 2D.
+        The columns are in the same order as in Table.domain._.
+        If writable == False (default), the numpy writable flag is set to false.
+        This means write operations on this array will loudly fail.
+
+        This is not the same as .values, but instead transforms the attributes to
+        numeric values (where possible); e.g. uses var.values indices instead of
+        actual descriptors.
+
+        Parameters
+        ----------
+        X : bool, default False
+            Whether to include the domain attributes in the result.
+        Y : bool, default False
+            Whether to include the domain class variables in the result.
+        meta : bool, default False
+            Whether to include the domain metas in the result.
+        writable : bool, default False
+            Whether to mark the resulting domain as writable.
+        Returns
+        -------
+        np.ndarray
+            The numpy array of the selected and transformed table data.
+        """
+        cols = []
+        cols += self.domain.attributes if X else []
+        cols += self.domain.class_vars if Y else []
+        cols += self.domain.metas if meta else []
+
+        # support using this in TableSeries, whose len gives the number of columns
+        n_rows = 1 if isinstance(self, SeriesBase) else len(self)
+
+        # preallocate result, we fill it in-place
+        # we need a more general dtype for metas (commonly strings),
+        # otherwise assignment fails later
+        result = np.zeros((n_rows, len(cols)), dtype=object if meta else None)
+        # effectively a double for loop, see if this is a bottleneck later
+        for i, col in enumerate(cols):
+            if isinstance(self, SeriesBase):
+                # if this is used in TableSeries, we don't have a series but an element
+                result[:, i] = col.to_val(self[col])
+            else:
+                vals = col.to_val(self[col]).values
+                result[:, i] = vals.values if hasattr(vals, 'values') else vals
+        result.setflags(write=writable)
+        return result
+
+    @property
+    def X(self):
+        """Return a read-only 2D numpy array of X.
+
+        The columns are in the same order as the columns in Table.domain.attributes.
+
+        Returns
+        -------
+        np.ndarray
+        """
+        return self._to_numpy(X=True)
+
+    @property
+    def Y(self):
+        """Return a read-only numpy array of Y.
+
+        If there is only one column, a one-dimensional array is returned. Otherwise 2D.
+        The columns are in the same order as the columns in Table.domain.class_vars.
+
+        Returns
+        -------
+        np.ndarray
+        """
+        result = self._to_numpy(Y=True)
+        return result[:, 0] if result.shape[1] == 1 else result
```

Comment:
No, fuck me, please lets have `.y` be the first class' values and `.Y` all of them, always 2D. For backwards compat, this only bears replacing all `.Y` calls (and idiotic `if` statements around them) with `.y`. I remember everyone agreeing to this before. @astaric @janezd